crates/subspace-farmer/src/farming.rs

```diff
@@ -61,13 +70,19 @@ impl Farming {
 }
 
 impl Drop for Farming {
+    /// by sending an empty message to the receiver, we can interrupt the process
     fn drop(&mut self) {
-        // we don't have to do anything in here
-        // this is for clarity and verbosity
+        // we actually don't have to do anything in here
+        // the default implementation will drop the sender automatically (without writing any code
+        // in here) and receiver will know that sender is dropped via an error
+        // we could handle that error, and achieve the same functionality (we actually do that too)
+        // but we choose to send an empty message instead, for clarity
         let _ = self.sender.take().unwrap().send(());
     }
 }
 
+/// races between the farming process and the receiver to check if the instance is dropped
+/// basically, makes the farming process interruptable
 async fn background_farming<T: RpcClient + Send>(
```

Comment:
If you move all of the contents of this function into `Farming::start()` and rename `receiver` to `stop_receiver`, you no longer need comments in it.

Reply:
All the other suggestions are implemented. But I'm skeptical on this one. 
`Farming::start()` looks very clean atm, and its job is to start and return the instance immediately. 
By separating the **racing** part into another function, I think the code is more modular and maintainable. And 2 lines of comment wouldn't hurt. 
For an outsider, I believe the current version is easier to comprehend.

Reply:
The issue is not that there are 2 comment lines, it is that they are unnecessary, code is shorter and easier to read if you don't have this function in the first place.

Reply:
Okay :)